lib/motors/controller.cpp

```diff
@@ -1,9 +1,105 @@
 #include "controller.hpp"
 
+#include <fstream>
+
 namespace hyped::motors {
 
-Controller::Controller(core::ILogger &logger) : logger_(logger)
+std::optional<Controller> Controller::create(core::ILogger &logger, std::string &message_file_path)
 {
+  std::ifstream input_stream(message_file_path);
+  if (!input_stream.is_open()) {
+    logger.log(core::LogLevel::kFatal, "Failed to open file %s", message_file_path.c_str());
+    return std::nullopt;
+  }
+  rapidjson::IStreamWrapper input_stream_wrapper(input_stream);
+  rapidjson::Document document;
```

Comment:
`const`s?

Reply:
the `IStreamWrapper` and `Document` can't be `const`, `ParseResult` can be